yjit/src/codegen.rs

```diff
@@ -5068,6 +5068,9 @@ enum SpecVal {
     BlockParamProxy,
     PrevEP(*const VALUE),
     PrevEPOpnd(Opnd),
+    // To avoid holding values across C calls for complex calls,
+    // we might need to set the SpecVal earlier in the call sequence
+    AlreadySet,
```

Comment:
This is one kind of `block_handler`, so I think this (and `BlockParamProxy`) should be a variant of `BlockHandler` enum. But feel free to leave it in this PR since I've been procrastinating on refactoring `BlockParamProxy` too.

Reply:
👍🏼 I'll do the refactor today. `SpecVal::None` should also go in `BlockHandler` as it puts `VM_BLOCK_HANDLER_NONE`.